### PR TITLE
refactor: remove wrapping `Result`

### PR DESCRIPTION
**Pull request by hirehamir**

### CODE REVIEW

File: src/platform_impl/x11/mod.rs [hunk 1]
```diff
@@ -26,10 +26,10 @@ pub struct GlobalHotKeyManager {
 }
 
 impl GlobalHotKeyManager {
-    pub fn new() -> crate::Result<Self> {
+    pub fn new() -> Self {
         let (thread_tx, thread_rx) = unbounded();
         std::thread::spawn(|| events_processor(thread_rx));
-        Ok(Self { thread_tx })
+        Self { thread_tx }
     }
```
Comment (hirehamir) [inline src/platform_impl/x11/mod.rs:33]: The `Result` here appears unnecessary and used.

This shouldn't cause any external changes. The goal is to make the API a bit cleaner.

**Comment by hirehamir** [comment]
`cargo test --all-features` passes locally.

**Comment by FabianLars** [comment]
The other platforms can return Errors though so this is required for the code to work on all platforms.

**Comment by hamirmahal** [comment]
Ah, I see; my mistake

**Pull request closed without merging**